### PR TITLE
Feature/test cloud environment variables

### PR DESCRIPTION
**Pull request by achocron**

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 1]
```diff
@@ -785,4 +802,25 @@ void unregister() {
...
+            }
+            String[] disableServicesList = disableServices.split(",");
+            for (String service : disableServicesList) {
+                service = service.trim();
+                if (service.equals(DISABLE_ALL_SERVICES) || service.equals(serviceName)) {
+                    return true;
+                }
+            }
+            return false;
+        } catch (NoClassDefFoundError | IllegalAccessError | IllegalStateException e) {
```
[the remark this answers is not in the document]
Comment (achocron) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: Actually i didn't copy from firebase, I originally didn't have it but it got thrown during a test so i added it

File: sdk/appcenter/build.gradle [hunk 2]
```diff
@@ -1,2 +1,8 @@
 description = 'This package contains the basic functionalities that all App Center services use to communicate with the backend.'
-evaluationDependsOn(':sdk')
\ No newline at end of file
+evaluationDependsOn(':sdk')
+
+dependencies {
+    def version = "1.0.1"
+    compileOnly "com.android.support.test:runner:${version}"
+    testCompile "com.android.support.test:runner:${version}"
```
[the remark this answers is not in the document]
Comment (achocron) [reply sdk/appcenter/build.gradle]: instead of testCompile?

[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter/build.gradle]: yes

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: Can you explain what is causing IllegalStateException here?

File: sdk/appcenter/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter/build.gradle]: also you can make the string a variable

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (achocron) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: I can't and I don't know how to reproduce. If don't see it happening, I can remove the catch for that one

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 3]
```diff
@@ -507,6 +522,8 @@ private final synchronized void startServices(Class<? extends AppCenterService>.
                     AppCenterService serviceInstance = (AppCenterService) service.getMethod("getInstance").invoke(null);
                     if (mServices.contains(serviceInstance)) {
                         AppCenterLog.warn(LOG_TAG, "App Center has already started the service with class name: " + service.getName());
+                    } else if (shouldDisable(serviceInstance.getServiceName())) {
+                        AppCenterLog.debug(LOG_TAG, "Environment variable to disable service has been set; not starting service " + service.getName() + ".");
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: Instrumentation variable

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.microsoft.appcenter.utils;
+
+import android.os.Bundle;
+import android.support.test.InstrumentationRegistry;
+
+/**
+ * Wraps InstrumentationRegistry class to enable mocking in unit tests.
+ */
+public class InstrumentationRegistryHelper {
+    public static Bundle getArguments() throws NoClassDefFoundError, IllegalAccessError {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java]: Add blank line plus Javadoc to avoid a warning.

**Pull request merged**